### PR TITLE
[hbase] Separate HBase v0.9x and v1.x bindings

### PR DESCRIPTION
**Pull request by cmccoy**
See #272.
This change breaks `HBaseClient10` into a separate module, and downgrades the HBase version used by the `hbase` binding to v0.98.13.

Note that the HBase 1.0 binding is now accessed via `hbase1`, instead of `hbase-10` so that the runner script maps to the correct binding.

I tested against HBase v0.98.13 (with `hbase`) and v1.0.1.1 (with `hbase1`).  The naming is a bit confusing...does anyone have an opinion on this naming shceme versus an `hbase09` and `hbase1` binding?

### CODE REVIEW

**Comment by busbey** [comment]
I'd like us to support HBase 0.94, and that won't work with the HBase 0.98 jars.

Do we want to have per-major-client-version bindings?

that would mean:

* HBase 0.94
* HBase 0.98
* HBase 1.y

That would mean we couldn't, e.g. take advantage of client-side optimizations in minor versions (atleast not without gymnastics). I certainly like this better than the alternative of per-minor-client-version bindings.

Unrelated, does keeping the 1.y API stuff in a module allow that module to build against the HBase 2.0.0-SNAPSHOT? Just wondering if we can close out #379 via this.

**Comment by busbey** [comment]
wrt naming, what if we introduce a means for bindings to have submodules and a way to tell us which class corresponds to a version request?

Essentially, something that would let us do:
```
$ ycsb load hbase --binding-version 1.0
```
or
```
$ ycsb load hbase --binding-version 0.94
```

instead of having it tied to the binding name directly.

**Comment by busbey** [comment]
the travis failure is because the distribution assembly pom needs to list the hbase1 binding as a dependency

**Comment by cmccoy** [comment]
Fixed the travis failure, thanks.  I'll try to see if this addresses #379 tomorrow.

Re: the `--binding-version` flag...how would we handle new (and backwards compatible) versions? Would there be some mapping from DB version to binding version?

**Comment by cmccoy** [comment]
OK, I separated the HBase bindings into three versions:

* HBase 1.x (`hbase10`)
* HBase 0.98.x (`hbase098`)
* HBase 0.94.x (`hbase094`)

The `hbase094` binding has no actual code, it just pulls in `hbase098` and replaces dependencies. I don't have a lot of experience with Maven; LMK if there's a tidier way of doing this.

I'd like to get this in and put off adding a `--binding-version` flag.

**Comment by busbey** [comment]
I'm fine waiting on the binding-version flag.

@gkamat do you have time to review this?

@cmccoy are you wanting to get this into the release this weekend?

**Comment by busbey** [comment]
as the patch currently stands, we'd have to make the next release version 0.4.0, since the "hbase" binding target goes away.

**Comment by cmccoy** [comment]
I'm happy targeting this for the next release (leaving it out of 0.3.1).

File: hbase10/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License. See accompanying
+LICENSE file.
+-->
+
+# HBase (1.x) Driver for YCSB
+This driver is a binding for the YCSB facilities to operate against a HBase 1.x Server cluster.
+To run against an HBase 0.9x cluster, use the `hbase` binding.
```
Comment (busbey) [inline hbase10/README.md]: I think with this patch there is no "hbase" binding, right?

File: hbase10/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+## Quickstart
+
+### 1. Start a HBase Server
+You need to start a single node or a cluster to point the client at. Please see [Apache HBase Reference Guide](http://hbase.apache.org/book.html) for more details and instructions.
+
+### 2. Set up YCSB
+You need to clone the repository and compile everything.
```
Comment (busbey) [inline hbase10/README.md]: since we have a new file here, could we stick with downloading instructions?

File: pom.xml [hunk 3]
```diff
@@ -69,7 +69,9 @@ LICENSE file.
   <properties>
     <maven.assembly.version>2.5.5</maven.assembly.version>
     <maven.dependency.version>2.10</maven.dependency.version>
-    <hbase.version>1.0.0</hbase.version>
+    <hbase094.version>0.94.27</hbase094.version>
+    <hbase098.version>0.98.13-hadoop2</hbase098.version>
+    <hbase10.version>1.0.1.1</hbase10.version>
```
Comment (busbey) [inline pom.xml]: 1.0.2 is out now, might as well update.

File: pom.xml [hunk 4]
```diff
@@ -69,7 +69,9 @@ LICENSE file.
   <properties>
     <maven.assembly.version>2.5.5</maven.assembly.version>
     <maven.dependency.version>2.10</maven.dependency.version>
-    <hbase.version>1.0.0</hbase.version>
+    <hbase094.version>0.94.27</hbase094.version>
+    <hbase098.version>0.98.13-hadoop2</hbase098.version>
```
Comment (busbey) [inline pom.xml]: 0.98.14 is out now, might as well update.

File: hbase10/README.md (hunk 1 above)
Comment (busbey) [inline hbase10/README.md]: this doc error is the only fix-before-push thing I see.

**Comment by cmccoy** [comment]
Sorry for the delay. All comments addressed - thanks!

**Comment by busbey** [comment]
+1

**Pull request merged**